cpp/src/copying/copy.cu

```diff
@@ -132,14 +131,26 @@ std::unique_ptr<column> copy_if_else( Left const& lhs,
 
    auto bool_mask_device_p = column_device_view::create(boolean_mask);
    column_device_view bool_mask_device = *bool_mask_device_p;                                    
-   
-   auto filter = [bool_mask_device] __device__ (cudf::size_type i) { return bool_mask_device.element<cudf::experimental::bool8>(i); };
-   return cudf::experimental::type_dispatcher(lhs.type(),                                             
-                                             copy_if_else_functor{},
-                                             lhs, rhs, boolean_mask.size(),
-                                             left_nullable, right_nullable,
-                                             filter,
-                                             mr, stream);
+  
+   if (boolean_mask.nullable() and boolean_mask.has_nulls()) { 
```

Comment:
You should only need to check for `has_nulls()` here. If there are nulls that means it also is nullable.

Reply:
has_nulls() directly starts null_count rather than looking if there is a null mask or not.

Reply:
Well `null_count` cannot be `>0` without a `null_mask`.

Reply:
I was just concerned about multiple calls, but it shouldn't be much of a burden.